source/Halibut/Transport/TcpConnectionFactory.cs

```diff
@@ -57,10 +59,11 @@ public async Task<IConnection> EstablishNewConnectionAsync(ExchangeProtocolBuild
             log.Write(EventType.OpeningNewConnection, $"Opening a new connection to {serviceEndpoint.BaseUri}");
 
             var certificateValidator = new ClientCertificateValidator(serviceEndpoint);
-            var client = await CreateConnectedTcpClientAsync(serviceEndpoint, halibutTimeoutsAndLimits, log, cancellationToken);
+            var client = await CreateConnectedTcpClientAsync(serviceEndpoint, halibutTimeoutsAndLimits, streamFactory, log, cancellationToken);
             log.Write(EventType.Diagnostic, $"Connection established to {client.Client.RemoteEndPoint} for {serviceEndpoint.BaseUri}");
             
-            var networkTimeoutStream = client.GetNetworkTimeoutStream();
+            var networkTimeoutStream = streamFactory.CreateStream(client);
```

Comment:
A small change I noticed is that here, we used to create the `NetworkTimeoutStream` always (regardless of async Halibut). Whereas now, we only create it if async Halibut is turned on.

I think it's ok, but wanted to draw your attention to it in case it's not 🙂

Reply:
I think in practice `EstablishNewConnectionAsync` is only called when async halibut is enabled so it should always be the same.

Reply:
Ah, of course. Disregard 😁